### DISCUSSION TITLE
I'd like configure IndefiniteWaitOneShotStartupCheckStrategy by properties, not only using Java class

### DISCUSSION THREAD

**Discussion by netstart** [testcontainers/testcontainers-java#4302] [Q&A] [answered]
I need use "Indefinite one shot startup strategy example" by testcontainer.properties.
How could I configure IndefiniteWaitOneShotStartupCheckStrategy using properties?

For real, I'd like to configure this three examples by properties, but I don't know how to do this:

```
container.withStartupCheckStrategy(new IndefiniteWaitOneShotStartupCheckStrategy());
```

```
container.waitingFor(Wait.forLogMessage("Completed: ALTER PLUGGABLE DATABASE XEPDB1 SAVE STATE", 1))
```

```
container.withStartupTimeoutSeconds(30);
```

Reference:
https://www.testcontainers.org/features/startup_and_waits/#indefinite-one-shot-startup-strategy-example

**Comment by kiview** [comment, marked as answer]
You can't configure a global wait strategy (if this is what you are looking for?) via properties. You have to configure the Java object. 
It's also unclear why you ever would want to configure this in a property, so if you give some context with regards to what you try to achieve, I might be able to give better advice.

**Reply by netstart** [reply]
Sometimes We use multiple database docker images that are slow to startup.

To use multiple database docker images on our Java Spring Framework application, we create multiple application.yml to define URL and driver-class-name, like this:

- application-oracle-test.yml
- application-mysql-test.yml

```
spring:
  datasource:
    url: jdbc:tc:oracle:thin:@localhost:1521
    driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
```


In our Continuous Delivery (Jenkins) we use an environment variable to enable or disable the execution of these tests.

Below you can see a test example.
```
@ActiveProfiles("oracle-test")
@DisabledIfSystemProperty(named = "test.database", matches = "(oracle|oracle-test)")
public class OracleLiquibaseTest extends AbstractIntegrationTest {

    @Autowired
    private JdbcTemplate jdbcTemplate;

    @Test
    public void testSimple() throws SQLException {
        assertEquals(jdbcTemplate.queryForObject("SELECT 1 FROM dual", Long.class), Long.valueOf(1L));
    }
}
```


When I need to execute this test locally, I execute something like this:

```
mvn test -Dtest.database=oracle-test
```

or


```
mvn test -Dtest.database=mysql-test
```

So, in this case, we liked access configuration to wait for container startup by testcontainer.properties, or by Spring Framework, in this case using application-oracle-test.yml
